app/src/main/java/edu/uw/myuw_android/NavDrawerMain.kt

```diff
@@ -0,0 +1,132 @@
+        })
+
+        UserInfoStore.email.observe(this, Observer {
+            navView.getHeaderView(0).findViewById<TextView>(R.id.drawer_email).text = it
+        })
+
+        navView.setNavigationItemSelectedListener {
+            Log.d("NavDrawerMain - onCreate", it.title.toString())
+            when (it.itemId) {
+                R.id.logout -> {
+                    Log.d("NavDrawerMain - logout", "logging out user")
+                    val currentState = UserInfoStore.readAuthState(this)
+                    val clearedState = AuthState(currentState.authorizationServiceConfiguration!!)
+                    if (currentState.lastRegistrationResponse != null) {
+                        clearedState.update(currentState.lastRegistrationResponse)
+                    }
+                    UserInfoStore.writeAuthState(this, clearedState)
+                    val mainIntent = Intent(this, LoginActivity::class.java)
+                    mainIntent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
+                    startActivity(mainIntent)
+                    finish()
+                    true
+                }
+                else -> {
+                    val handled = NavigationUI.onNavDestinationSelected(it, navController)
+                    if (handled) drawerLayout.closeDrawer(navView)
+                    handled
+                }
+            }
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        menuInflater.inflate(R.menu.nav_drawer_main, menu)
+
+        // Get the SearchView and set the searchable configuration
+        val searchManager = getSystemService(Context.SEARCH_SERVICE) as SearchManager
+        (menu.findItem(R.id.action_open_search).actionView as SearchView).apply {
+            // Assumes current activity is the searchable activity
+            setSearchableInfo(searchManager.getSearchableInfo(componentName))
+        }
+
+        return true
+    }
+
+    override fun onSupportNavigateUp(): Boolean {
+        val navController = findNavController(R.id.nav_host_fragment)
+        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        Log.d("NavDrawerMain - onOptionsItemSelected", item.toString())
+        return when (item.itemId) {
+            R.id.action_open_search -> {
+                startActivity(Intent(this, SearchActivity::class.java))
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    override fun onBackPressed() {
+        val navHostFragment = supportFragmentManager.findFragmentById(R.id.nav_host_fragment) as NavHostFragment
+        val currentFragment = navHostFragment.childFragmentManager.fragments[0] as CommonWebViewFragment
+        if (currentFragment.onBackPress()) super.onBackPressed()
+    }
```

Comment:
What is the expected behavior of this? Are we trying to close the Nav Menu if opened when back is pressed? But currently when I press back button with the Nav menu opened, only the WebView is reacting to it which looks like a bug to me.

Reply:
This was part of an older design that I forgot to remove. It was suppose to let you navigate inside a webview which is no longer required. I will remove this.

Reply:
I removed it